wpilibc/shared/src/Commands/CommandGroup.cpp

```diff
@@ -158,21 +153,20 @@ void CommandGroup::AddParallel(Command* command, double timeout) {
     return;
   }
 
+  m_commands.emplace_back(command, CommandGroupEntry::kSequence_BranchChild,
+                          timeout);
+
   command->SetParent(this);
 
-  m_commands.push_back(CommandGroupEntry(
-      command, CommandGroupEntry::kSequence_BranchChild, timeout));
   // Iterate through command->GetRequirements() and call Requires() on each
   // required subsystem
-  Command::SubsystemSet requirements = command->GetRequirements();
-  auto iter = requirements.begin();
-  for (; iter != requirements.end(); iter++) Requires(*iter);
+  for (auto requirement : command->GetRequirements()) Requires(requirement);
 }
 
 void CommandGroup::_Initialize() { m_currentCommandIndex = -1; }
 
 void CommandGroup::_Execute() {
-  CommandGroupEntry entry;
+  CommandGroupEntry* entry;
```

Comment:
It avoids unnecessarily copying elements of m_commands, which are of type CommandGroupEntry. Pointers had to be used because reassigning references is not allowed.